frontend/lib/features/home/home_page.dart

```diff
@@ -1,21 +1,169 @@
+  Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(title: Text(Loc.of(context).home)),
-      body: const Column(
-        crossAxisAlignment: CrossAxisAlignment.stretch,
+      body: SafeArea(
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            buildAccountBalance(context),
+            Padding(
+              padding: const EdgeInsets.symmetric(
+                  horizontal: Grid.side, vertical: Grid.xs),
+              child: Text(
+                Loc.of(context).activity,
+                style: Theme.of(context).textTheme.bodyMedium,
+              ),
+            ),
+            Expanded(
+              child: txns.isEmpty
+                  ? buildEmptyState(context)
+                  : ListView(
+                      children: txns.map((txn) {
+                        return ListTile(
+                          title: Text(
+                            txn.title,
+                            style: Theme.of(context)
+                                .textTheme
+                                .titleSmall
+                                ?.copyWith(
+                                  fontWeight: FontWeight.bold,
+                                ),
+                          ),
+                          // TODO: display status from txn
+                          subtitle: Text(
+                            'status',
+                            style: Theme.of(context)
+                                .textTheme
+                                .labelMedium
+                                ?.copyWith(
+                                  fontWeight: FontWeight.w300,
+                                ),
+                          ),
+                          trailing: Text(txn.amount.toString()),
+                          leading: Container(
+                            width: Grid.lg,
+                            height: Grid.lg,
+                            decoration: BoxDecoration(
+                              color:
+                                  Theme.of(context).colorScheme.surfaceVariant,
+                              shape: BoxShape.circle,
+                            ),
+                            // TODO: use $ or first letter of name based on txn type
+                            child: const Center(
+                              child: Text('\$'),
+                            ),
+                          ),
+                        );
+                      }).toList(),
+                    ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget buildAccountBalance(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: Grid.side),
+      child: Container(
+        decoration: BoxDecoration(
+          border: Border.all(
+              color: Theme.of(context).colorScheme.outline, width: 2.0),
+          borderRadius: BorderRadius.circular(25.0),
```

Comment:
Do we want a standard radius for these? We could use the grid or define a similar radius size somewhere.